### PR TITLE
Return of the Charcoal Nerf (Hobo returning to Coder Status V.2)

### PR DESCRIPTION
**Pull request by Therandomhoboo**
# Document the changes in your pull request

Co-Authored by @Majkl-J 

Stab number 2 at the charcoal nerf. So I've had some words with peeps in discord and come to the conclusion of reducing its effectiveness and increasing the effectiveness of Calomel so it gets used more despite it giving toxin damage.
[EDIT]
So, spoke with Molti about this and then Waterpig and went with the idea of making signal for purge itself which should in future if we put any more purge in (like Halopedriodoldolod) or others, then we can use this register the purging itself.

It is sad to say that I will be removing the signals for the time being just until we can work another way around it. For the time being I want this out so at least something is done

So as following:
Charcoal: 1 purge rate reduce to 0.5
Calomel: Purge rate increased to 5 from 2.5

[Removed for the time being]
A Signal for Adding mob_reagents & removing said reagents been added to the DEFINE_, Reagents
Chemistry.DM has the send signals there.
Calomel has the use of the signals

# Why is this good for the game?

Stops charcoal hoarding, makes chem make better purges, makes poisons, chems and drinks more viable early game than straight up getting told no because charcoal dumped it all out before it even tickled.

# Testing

Yes, anytime Calomel fully purges a chem, it will apply the tox damage.
Charcoal purge is slower but this makes it feel like a roundstart chem.

# Wiki Documentation

Change Charcoal effectiveness to 0.5, toxin heal to 3.5, Calomel purge to 5 instead of 2.5

# Changelog

:cl:  

tweak: Charcoal purge rate reduced to 0.5 from 1
tweak: Calomel Purge rate increased to 5 from 2.5
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 1]
```diff
@@ -529,8 +528,8 @@
 /datum/reagent/medicine/calomel/on_mob_life(mob/living/carbon/M)
 	for(var/datum/reagent/R in M.reagents.reagent_list)
 		if(R != src)
-			M.reagents.remove_reagent(R.type,2.5)
-	if(M.health > 20)
+			M.reagents.remove_reagent(R.type,5)
+	if(M.reagents.remove_reagent/R == TRUE)
```
Comment (Moltijoe) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: do you uhh, want help with that?

File: code/modules/reagents/chemistry/reagents.dm [hunk 2]
```diff
@@ -112,13 +112,16 @@ GLOBAL_LIST_INIT(name2reagent, build_name2reagent())
 
 ///Called after a reagent is transfered
 /datum/reagent/proc/on_transfer(atom/A, methods=TOUCH, trans_volume)
+
 /// Called when this reagent is first added to a mob
 /datum/reagent/proc/on_mob_add(mob/living/L)
-	return
+    SEND_SIGNAL(L, COMSIG_MOB_ADD_REAGENT, src.type)
+    return
```
Comment (Maia-J) [inline code/modules/reagents/chemistry/reagents.dm]:
```suggestion
    SEND_SIGNAL(L, COMSIG_MOB_ADD_REAGENT, src)
    return
```

File: code/modules/reagents/chemistry/reagents.dm [hunk 3]
```diff
@@ -112,13 +112,16 @@ GLOBAL_LIST_INIT(name2reagent, build_name2reagent())
...
 
 /// Called when this reagent is removed while inside a mob
 /datum/reagent/proc/on_mob_delete(mob/living/L)
-	return
+    SEND_SIGNAL(L, COMSIG_MOB_REMOVE_REAGENT, src.type)
+    return
```
Comment (Maia-J) [inline code/modules/reagents/chemistry/reagents.dm]:
```suggestion
    SEND_SIGNAL(L, COMSIG_MOB_REMOVE_REAGENT, src)
    return
```

**Review by Majkl-J — CHANGES_REQUESTED** [review]
I fucked up here, just send the src, more versatile

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 4]
```diff
@@ -529,12 +528,27 @@
...
+/datum/reagent/medicine/calomel/on_mob_remove(mob/living/L)
+    UnregisterSignal(L, COMSIG_MOB_REMOVE_REAGENT)
+    return ..()
+
+/datum/reagent/medicine/calomel/proc/reagent_purged(datum/source, datum/reagent/purged_chem)
+    // Source is the mob, so do this:
+    var/mob/living/affected_mob = source
+    if(!istype(affected_mob))
+		return
+		affected.adjustToxLoss(2.5*REM, 0)
```
Comment (Maia-J) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]:
```suggestion
	affected.adjustToxLoss(2.5*REM, 0)
```

**Review by Majkl-J — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 5]
```diff
@@ -523,18 +522,33 @@
...
 		. = 1
 	..()
 
+/datum/reagent/medicine/calomel/on_mob_add(mob/living/L)
+	RegisterSignal(L,	COMSIG_MOB_REMOVE_REAGENT,	PROC_REF(reagent_purged))
+	return	..()
+
+/datum/reagent/medicine/calomel/on_mob_remove(mob/living/L)
+	UnregisterSignal(L,	COMSIG_MOB_REMOVE_REAGENT)
+	return	..()
```
Comment (Maia-J) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]:
```suggestion
/datum/reagent/medicine/calomel/on_mob_delete(mob/living/L)
	UnregisterSignal(L, COMSIG_MOB_REMOVE_REAGENT)
	return ..()
```

**Review by Majkl-J — CHANGES_REQUESTED** [review]
Woops, again

**Comment by Therandomhoboo** [comment]
> Woops, again

Kek

**Comment by Therandomhoboo** [comment]
So with how it works now. You get the tox damage after the chemical that it is purging is finished. With that in mind, I think personally raising tox damage a little would be best but as of current it does work.

**Review by Moltijoe — CHANGES_REQUESTED** [review]
This can be weaponized to kill someone in seconds
We've already discussed this, just requesting changes so someone like manatee cant merge

**Review by Moltijoe — APPROVED** [review]

**Pull request merged**